src/eventsub/event.rs

```diff
@@ -588,42 +535,111 @@ impl Event {
-            channel::ChannelRaidV1;
-            channel::ChannelSubscriptionEndV1;
-            channel::ChannelSubscriptionGiftV1;
-            channel::ChannelSubscriptionMessageV1;
-            channel::ChannelGoalBeginV1;
-            channel::ChannelGoalProgressV1;
-            channel::ChannelGoalEndV1;
-            channel::ChannelHypeTrainBeginV1;
-            channel::ChannelHypeTrainProgressV1;
-            channel::ChannelHypeTrainEndV1;
-            stream::StreamOnlineV1;
-            stream::StreamOfflineV1;
-            user::UserUpdateV1;
-            user::UserAuthorizationGrantV1;
-            user::UserAuthorizationRevokeV1;
-        })
+        Ok(fill_events!(match_event()))
+    }
+
+    /// Parse a websocket frame as an [`EventsubWebsocketData`]
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use twitch_api::eventsub::{Event, EventsubWebsocketData};
+    /// let notification = r#"
+    /// {
+    ///     "metadata": {
+    ///         "message_id": "befa7b53-d79d-478f-86b9-120f112b044e",
+    ///         "message_type": "notification",
+    ///         "message_timestamp": "2019-11-16T10:11:12.123Z",
+    ///         "subscription_type": "channel.follow",
+    ///         "subscription_version": "1"
+    ///     },
+    ///     "payload": {
+    ///         "subscription": {
+    ///             "id": "f1c2a387-161a-49f9-a165-0f21d7a4e1c4",
+    ///             "status": "enabled",
+    ///             "type": "channel.follow",
+    ///             "version": "1",
+    ///             "cost": 1,
+    ///             "condition": {
+    ///                 "broadcaster_user_id": "12826"
+    ///             },
+    ///             "transport": {
+    ///                 "method": "websocket",
+    ///                 "session_id": "AQoQexAWVYKSTIu4ec_2VAxyuhAB"
+    ///             },
+    ///             "created_at": "2019-11-16T10:11:12.123Z"
+    ///         },
+    ///         "event": {
+    ///             "user_id": "1337",
+    ///             "user_login": "awesome_user",
+    ///             "user_name": "Awesome_User",
+    ///             "broadcaster_user_id": "12826",
+    ///             "broadcaster_user_login": "twitch",
+    ///             "broadcaster_user_name": "Twitch",
+    ///             "followed_at": "2020-07-15T18:16:11.17106713Z"
+    ///         }
+    ///     }
+    /// }
+    /// "#;
+    /// let event: EventsubWebsocketData<'_> = Event::parse_websocket(notification)?;
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    #[cfg(feature = "unsupported")]
+    pub fn parse_websocket(frame: &str) -> Result<EventsubWebsocketData<'_>, PayloadParseError> {
+        #[derive(Deserialize)]
+        #[cfg_attr(feature = "deny_unknown_fields", serde(deny_unknown_fields))]
+        struct EventsubWebsocketFrame<'a> {
+            metadata: EventsubWebsocketMetadata<'a>,
+            #[serde(borrow)]
+            payload: &'a serde_json::value::RawValue,
+        }
```

Comment:
I think this should be exposed to the user if they want to (for whatever reason) implement the parsing on their own.

Reply:
you mean exposing this struct? It would be rather simple to do it yourself, since you'd just pretty much the same thing but without the parsing done in this method. ie

```rust
#[derive(Deserialize)]
struct EventsubWebsocketFrame<'a> {
    metadata: twitch_api::eventsub::EventsubWebsocketMetadata<'a>,
    payload: WhateverTypeINeed,
}
let data: EventsubWebsocketFrame = crate::parse_json(frame, true)?;
```

do you think that works?

sidenote:

The metadata is exposed in `Event::parse_websocket` through `EventsubWebsocketData`s metadata fields. Could maybe add a getter for it.

Reply:
> do you think that works?

Yeah that's fair.

> The metadata is exposed in `Event::parse_websocket` through `EventsubWebsocketData`s metadata fields. Could maybe add a getter for it.

That should be enough I think.